polygeist/test/polygeist-opt/sycl/sycl-raise-host.mlir

```diff
@@ -818,3 +818,223 @@ llvm.func @raise_set_captured(%handler: !llvm.ptr) {
   
   llvm.return
 }
+
+// -----
+
+// COM: Check raising of set_nd_range, set_captured and set_kernel to schedule_kernel.
+//      Note that the raising pattern is currently limited to `parallel_for`-style
+//      launches with at least one argument.
+
+gpu.module @device_functions {
+  gpu.func @foo() kernel {
+    gpu.return
+  }
+}
+
+!lambda_class = !llvm.struct<"class.lambda", opaque>
+!sycl_accessor_1_21llvm2Evoid_rw_gb = !sycl.accessor<[1, !llvm.void, read_write, global_buffer], (!llvm.void)>
+!kernel_signatures_type = !llvm.array<7 x struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)>>
+
+llvm.mlir.global internal constant @_ZN4sycl3_V16detailL17kernel_signaturesE() : !kernel_signatures_type {
+  %zero = llvm.mlir.constant(0 : i32) : i32
+  %dummy = llvm.mlir.constant(-987654321 : i32) : i32
+  %kind_accessor = llvm.mlir.constant(0 : i32) : i32
+  %kind_std_layout = llvm.mlir.constant(1 : i32) : i32
+  %kind_invalid = llvm.mlir.constant(15 : i32) : i32
+  %undef = llvm.mlir.undef : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)>
+  
+  // COM: accessor arg
+  %accessor_0 = llvm.insertvalue %kind_accessor, %undef[0] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  %accessor_1 = llvm.insertvalue %zero, %accessor_0[1] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  %accessor = llvm.insertvalue %zero, %accessor_1[2] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+
+  // COM: std_layout arg
+  %std_layout_0 = llvm.insertvalue %kind_std_layout, %undef[0] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  %std_layout_1 = llvm.insertvalue %zero, %std_layout_0[1] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  %std_layout = llvm.insertvalue %zero, %std_layout_1[2] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+
+  // COM: sentinel
+  %sentinel_0 = llvm.insertvalue %kind_invalid, %undef[0] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  %sentinel_1 = llvm.insertvalue %dummy, %sentinel_0[1] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  %sentinel = llvm.insertvalue %dummy, %sentinel_1[2] : !llvm.struct<"struct.sycl::_V1::detail::kernel_param_desc_t", (i32, i32, i32)> 
+  
+  // COM: mockup 3 kernel signatures here:
+  //      - accessor, sentinel
+  //      - std_layout, accessor
+  //      - std_layout, std_layout, std_layout
```

Comment:
I though we were execting a sentinel to be last

Reply:
Fixed.